### PR TITLE
added cmdlets for logger, properties, git and openIdConnect

### PR DESCRIPTION
**Pull request by solankisamir**
- Added cmdlets for Logger 
- Added cmdlets for Properties
- Added cmdlets for Git
- Added cmdlets for OpenIdConnect
- Fixed issues with Doc generation due to rename of namespace to AzureRm

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@solankisamir__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (sasolank). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/ApiManagementClient.cs [hunk 1]
```diff
@@ -1291,5 +1334,410 @@ public void AuthorizationServerRemove(PsApiManagementContext context, string ser
             Client.AuthorizationServers.Delete(context.ResourceGroupName, context.ServiceName, serverId, "*");
         }
         #endregion
+
+        #region Loggers
+        public PsApiManagementLogger LoggerCreate(
```
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/ApiManagementClient.cs]: No comments for these public methods?

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands.ApiManagement.ServiceManagement.csproj [hunk 2]
```diff
@@ -66,9 +66,9 @@
       <HintPath>..\..\..\packages\Microsoft.Azure.KeyVault.Core.1.0.0\lib\net40\Microsoft.Azure.KeyVault.Core.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <Reference Include="Microsoft.Azure.Management.ApiManagement, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+    <Reference Include="Microsoft.Azure.Management.ApiManagement">
```
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands.ApiManagement.ServiceManagement.csproj]: Do not remove the assembly evidence, please,  This can cause issues at runtime.

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/AddAzureApiManagementApiToProduct.cs [hunk 3]
```diff
@@ -1 +1 @@
...
        {
            Client.ApiAddToProduct(Context, ProductId, ApiId);

            if (PassThru)
            {
                WriteObject(true);
            }
        }
    }
}
```
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/AddAzureApiManagementApiToProduct.cs]: Remove this change

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/AzureApiManagementCmdletBase.cs [hunk 4]
```diff
@@ -65,5 +68,64 @@ protected virtual void HandleException(Exception ex)
         {
             WriteExceptionError(ex);
         }
+        
+        protected void WriteProgress(TenantConfigurationLongRunningOperation operation)
+        {
+            WriteProgress(new ProgressRecord(0, operation.OperationName, operation.Status.ToString()));
+        }
+
+        protected TenantConfigurationLongRunningOperation WaitForOperationToComplete(TenantConfigurationLongRunningOperation longRunningOperation)
```
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/AzureApiManagementCmdletBase.cs]: WHy is this not using the standard long running operation code?

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/GetAzureApiManagementGroup.cs [hunk 5]
```diff
@@ -15,11 +15,12 @@
...
 {
     using System;
+    using System.Collections.Generic;
     using System.Management.Automation;
     using Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Models;
 
-    [Cmdlet(VerbsCommon.Get, "AzureRmApiManagementGroup", DefaultParameterSetName = GetAll)]
-    [OutputType(typeof(PsApiManagementGroup))]
+    [Cmdlet(VerbsCommon.Get, Constants.ApiManagementGroup, DefaultParameterSetName = GetAll)]
+    [OutputType(typeof(IList<PsApiManagementGroup>))]
```
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/GetAzureApiManagementGroup.cs]: If this cmdlets can return both a single object and a list of objects, you should include both in the OutputType attributes.

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/GetAzureRmApiManagementTenantGitAccess.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+//  limitations under the License.
+
+namespace Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Commands
+{
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Models;
+
+    [Cmdlet(VerbsCommon.Get, Constants.ApiManagementTenantGitAccess)]
+    [OutputType(typeof(PsApiManagementAccessInformation))]
+    public class GetAzureRmApiManagementTenantGitAccess : AzureApiManagementCmdletBase
```
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/GetAzureRmApiManagementTenantGitAccess.cs]: Shouldn't this be GiAaccessInformation, or GitInformation?

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/NewAzureApiManagementApi.cs [hunk 7]
```diff
@@ -58,8 +58,9 @@ public class NewAzureApiManagementApi : AzureApiManagementCmdletBase
 
         [Parameter(
             ValueFromPipelineByPropertyName = true, 
-            Mandatory = false, 
-            HelpMessage = "Web API Path. Last part of the API's public URL. This URL will be used by API consumers for sending requests to the web service. Must be 1 to 400 characters long. This parameter is optional. Default value is $null.")]
+            Mandatory = true, 
+            HelpMessage = "Web API Path. Last part of the API's public URL. This URL will be used by API consumers for sending requests to the web service. Must be 1 to 400 characters long. This parameter is required.")]
```
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/NewAzureApiManagementApi.cs]: Line break

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/NewAzureApiManagementOpenIdConnectProvider.cs [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            HelpMessage = "Instance of PsApiManagementContext. This parameter is required.")]
+        [ValidateNotNullOrEmpty]
+        public PsApiManagementContext Context { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "Identifier of new openIdConnectProvider. This parameter is optional. If not specified will be generated.")]
```
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/NewAzureApiManagementOpenIdConnectProvider.cs]: Line breaks

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/NewAzureApiManagmentProperty.cs [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            HelpMessage = "Instance of PsApiManagementContext. This parameter is required.")]
+        [ValidateNotNullOrEmpty]
+        public PsApiManagementContext Context { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "Identifier of new property. This parameter is optional. If not specified will be generated.")]
```
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/NewAzureApiManagmentProperty.cs]: Line breaks

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/PublishAzureApiManagementTenantConfiguration.cs [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+            Mandatory = true,
+            HelpMessage = "Name of the Git branch from which the configuration is to be deployed to the configuration database. This parameter is required.")]
+        [ValidateNotNullOrEmpty]
+        public String Branch { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "Enforce deleting subscriptions to products that are deleted in this update. This parameter is optional. Default value is false.")]
+        public bool? Force { get; set; }
```
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/PublishAzureApiManagementTenantConfiguration.cs]: Force should be reserved for a SwitchParameter that short-circuits a call to ShouldContinue.  If this parameter is for some other purpose, you should call it something else.  And it should be a swtch parameter, regardless.

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementLogger.cs [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "Description of the Logger. This parameter is optional.")]
+        public String Description { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "Determines whether the records in the logger are buffered before publishing. This parameter is optional and the default value is true. ")]
+        public bool? IsBuffered { get; set; }
```
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementLogger.cs]: SwitchParameter is preferred, unless this is a value youreally expect to come form the pipeline.

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureRmApiManagementTenantGitAccess.cs [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+            HelpMessage = "Instance of PsApiManagementContext. This parameter is required.")]
+        [ValidateNotNullOrEmpty]
+        public PsApiManagementContext Context { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            HelpMessage = "Enable Git access. Set to true for enabling and false for disabling. This parameter is required.")]
+        [ValidateNotNullOrEmpty]
+        public bool Enabled { get; set; }
```
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureRmApiManagementTenantGitAccess.cs]: Switch parameter?  Or is this really coming from the pipeline?

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/AzureApiManagementCmdletBase.cs (hunk 4 above)
Comment (solankisamir) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/AzureApiManagementCmdletBase.cs]: There is a bug and it will take 2 weeks to go out, found out during powershell testing.

File: setup/azurecmdfiles.wxi [hunk 13]
```diff
@@ -204,6 +204,9 @@
                         <Component Id="cmp048B605B1072F4A5467F2AFFD8ED2459" Guid="*">
```
Comment (markcowl) [inline setup/azurecmdfiles.wxi]: Please revert these changes.  None of the dll-Help.psd1 files should be part of the installer package.

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/AzureApiManagementCmdletBase.cs (hunk 4 above)
Comment (markcowl) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/AzureApiManagementCmdletBase.cs]: Please provide a link to the bug.

**Comment by markcowl** [comment]
@azuresdkci add to whitelist

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/AzureApiManagementCmdletBase.cs (hunk 4 above)
Comment (solankisamir) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/AzureApiManagementCmdletBase.cs]: Its a service side Code bug, I will add a TODO to revert to standard long running operation code

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/GetAzureApiManagementGroup.cs (hunk 5 above)
Comment (solankisamir) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/GetAzureApiManagementGroup.cs]: Fixed for all cmdlets returning both single and list of objects.

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/NewAzureApiManagementApi.cs (hunk 7 above)
Comment (solankisamir) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/NewAzureApiManagementApi.cs]: Fixed

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/NewAzureApiManagementOpenIdConnectProvider.cs (hunk 8 above)
Comment (solankisamir) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/NewAzureApiManagementOpenIdConnectProvider.cs]: Fixed

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/NewAzureApiManagmentProperty.cs (hunk 9 above)
Comment (solankisamir) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/NewAzureApiManagmentProperty.cs]: Fixed

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/PublishAzureApiManagementTenantConfiguration.cs (hunk 10 above)
Comment (solankisamir) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/PublishAzureApiManagementTenantConfiguration.cs]: Changed to SwitchParameter with same Name.

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/GetAzureRmApiManagementTenantGitAccess.cs (hunk 6 above)
Comment (solankisamir) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/GetAzureRmApiManagementTenantGitAccess.cs]: No.. we use the same return object for two cmdlets

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureRmApiManagementTenantGitAccess.cs (hunk 12 above)
Comment (solankisamir) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureRmApiManagementTenantGitAccess.cs]: Changed to not coming from pipeline.

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementLogger.cs (hunk 11 above)
Comment (solankisamir) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementLogger.cs]: Fixed

File: setup/azurecmdfiles.wxi (hunk 13 above)
Comment (solankisamir) [inline setup/azurecmdfiles.wxi]: reverted the changes

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands.ApiManagement.ServiceManagement.csproj (hunk 2 above)
Comment (solankisamir) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands.ApiManagement.ServiceManagement.csproj]: Fixed this.

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/PublishAzureApiManagementTenantConfiguration.cs (hunk 10 above)
Comment (solankisamir) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/PublishAzureApiManagementTenantConfiguration.cs]: Fixed

File: src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementLogger.cs (hunk 11 above)
Comment (solankisamir) [inline src/ResourceManager/ApiManagement/Commands.ApiManagement.ServiceManagement/Commands/SetAzureApiManagementLogger.cs]: Fixed

**Comment by markcowl** [comment]
on demand run here:  http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/724/

signing run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/395/

**Comment by markcowl** [comment]
@solankisamir merge conflicts.  Please update

**Comment by solankisamir** [comment]
Merged from latest azure/dev 

Powershell On-Demand Run
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/737/console

Powershell Sigin-In Run
http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/404/console
